mygame.js

```diff
@@ -25,4 +25,111 @@ Scene.prototype.bug = function scene_bug(message) {
   } else {
     throw new Error(this.lineMsg() + message);
   }
-}
\ No newline at end of file
+}
+
+/*  AUTOMATIC CHOICE SELECTION #1
+    ---------------------------------
+    As with bug, we want to be able to be able to 'catch', but also
+    test and interact with *choice(s) programatically.
+
+    Example usage:
+
+    *comment configure automatic choice selection
+    *set cslib_catch_choice true
+    *comment dictate which options (comma separated, for nested choices) should be selected
+    *set cslib_choice_select "3"
+    *fake_choice
+      # Go east
+        Eastward bound.
+      # Go west
+        Westward bound.
+      # Go south
+        Southward bound.
+    *comment 'Southward bound.' will be printed here.
+    *comment And helper variables will be populated:
+    ${cslib_choice_count} == 3
+    ${cslib_choice_1} == "Go east."
+    ${cslib_choice_2} == "Go west."
+    *comment etc.
+    *comment and turn off automatic choice selection
+    *set cslib_catch_choice false
+*/
+
+
+/*  AUTOMATIC CHOICE SELECTION #2
+    ---------------------------------
+    The core functionality is implemented by overriding the renderOptions
+    method, which serves as the initial trigger to print the radio button
+    UI to the screen. We guard the old and new (automatic) functionality
+    on a global: cslib_catch_choice.
+
+    cslib_choice_select is another global that should contain the number
+    of the choice option to select (1 based), or a series of numbers,
+    separated by a comma, if your choices are nested.
+
+    A few helper variables are also created to help confirm that the length/number
+    (cslib_choice_count) of options and their text content (cslib_choice_n) is as expected.
+*/
+Scene.prototype.renderOptions = function renderOptions(groups, options, callback) {
+  var self = this;
+  for (var i = 0; i < options.length; i++)
+    options[i].name = self.replaceVariables(options[i].name);
+
+  if (this.stats.cslib_catch_choice) {
+    // automated choice selection for cslib_menu tests
+    if (this.stats.cslib_choice_select.length === 0)
+      throw new Error("Error: stuck in a choice menu but selections (cslib_choice_select) were exhausted.");
+
+    // Extract the next choice selection
+    var selections = this.stats.cslib_choice_select.split(",");
```

Comment:
do you need to support nested choices in your tests?

Reply:
Nested is probably the wrong word... 'subsequent' (though that would include nested).
But yes, this way I can check pagination works (both in terms of navigation, and that pages past page 1 are correct).
It's actually really quite fun...